stores/tasks.ts

```diff
@@ -30,64 +33,92 @@ export const useTasksStore = defineStore('tasks', () => {
     return orderedTasks
   })
 
-  watchEffect(() => {
-    if (!isReady.value && tasks.value) {
-      isReady.value = true
-    }
-  })
-
-  // Initialize task priority after first fetch
-  watchEffect(() => {
-    if (permissions.canEditLabels && isReady.value) {
-      initializePriority()
-    }
-  })
-
-  // Merge task-derived columns with existing columns
-  watchEffect(() => {
-    if (tasksByColumn.value) {
-      board.mergeColumns(Object.keys(tasksByColumn.value))
-    }
-  })
-
-  async function initializePriority() {
-    if (!tasks.value) {
-      return
+  const tasksWithoutPriority = computed(() => {
+    if (!permissions.canEditLabels || !tasks.value) {
+      return null
     }
 
     const tasksWithoutPriority: Task[] = []
-    let highestPriority = 0
 
     for (const task of tasks.value) {
       if (task.rpb.priority === null) {
         tasksWithoutPriority.push(task)
-      } else if (task.rpb.priority > highestPriority) {
-        highestPriority = task.rpb.priority
       }
     }
 
-    for (const [index, task] of tasksWithoutPriority.entries()) {
-      const priority = highestPriority + (index + 1) * taskPriorityIncrement
+    return tasksWithoutPriority
+  })
 
-      await updateTaskLabels(task, [...task.labels, createDataLabel('priority', priority)])
-    }
+  const { mutate: initializePriority, isPending: isInitializePriorityPending } = useMutation({
+    async mutationFn() {
+      if (!tasksByColumn.value || !tasksWithoutPriority.value) {
+        return
+      }
 
-    if (tasksWithoutPriority.length > 0) {
-      await refreshTasks()
-    }
-  }
+      function getHighestPriorityInColumn(column: string) {
+        const columnTasks = tasksByColumn.value?.[column]
+        if (!columnTasks) {
+          return 0
+        }
+
+        return columnTasks.reduce((acc, task) => {
+          return task.rpb.priority !== null && task.rpb.priority > acc
+            ? task.rpb.priority
+            : acc
+        }, 0)
+      }
+
+      const highestPriorityByColumn: Record<string, number> = {}
+      const taskIndexByColumn: Record<string, number> = {}
+
+      for (const task of tasksWithoutPriority.value) {
+        const { column } = task.rpb
+
+        if (highestPriorityByColumn[column] === undefined) {
```

Comment:
The argument of not using due to handling a "rogue" zero may not change much because it means we're already in a bad state since that should never be a value in this case. I've we're concerned that zero might come up we need to handle that properly in a different way. I'm resolving anyway, but I still think we should prefer the suggested syntax.